modules/http4s/src/smithy4s/http4s/EndpointBasedMiddleware.scala

```diff
@@ -0,0 +1,26 @@
+package smithy4s
+package http4s
+
+import org.http4s.HttpApp
+
+// format: off
+trait EndpointSpecificMiddleware[Alg[_[_, _, _, _, _]], F[_]] {
+  def prepare(service: Service[Alg])(
+      endpoint: Endpoint[service.Operation, _, _, _, _, _]
+  ): HttpApp[F] => HttpApp[F]
+}
+// format: on
```

Comment:
Really not sure about this interface. Could use some guidance/thoughts as to my usages of Alg, Op, and existential types here.

Reply:
The interface is okay in that it gives the correct amount of flexibility. I think the advertised use in your example is more problematic

Reply:
Ah right, I see what you are saying. I will add that and update the example. Thanks!